lib/full-site-editing/block-templates.php

```diff
@@ -386,41 +386,43 @@ function gutenberg_get_block_template( $id, $template_type = 'wp_template' ) {
  * @param string $post_type     Post type.
  * @return string The original, desired slug.
  */
-function gutenberg_filter_wp_template_unique_post_slug( $slug, $post_ID, $post_status, $post_type ) {
-	if ( 'wp_template' !== $post_type || 'wp_template_part' !== $post_type ) {
+function gutenberg_filter_wp_template_unique_post_slug( $placeholder, $slug, $post_ID, $post_status, $post_type ) {
+	if ( 'wp_template' !== $post_type && 'wp_template_part' !== $post_type ) {
 		return $slug;
 	}
 
 	// Template slugs must be unique within the same theme.
-	$theme = get_the_terms( $post_ID, 'wp_theme' )[0]->slug;
+	$theme = wp_get_theme()->get_stylesheet();
```

Comment:
At this point, no taxonomy exists for the new item.  Thus `get_the_terms` was returning `false`, and therefore an error/warning of us trying to treat `false` as an array with the `[0]->slug`.

Reply:
This will work now but won't scale later when we'll support multi-themes. Do you think we can still keep the `wp_unique_post_slug` filter instead of `pre` in which case the taxonomy will be there?

> We need to do this on the pre-* hook. Otherwise, it executes too late (even on 1st priority), and the slug it receives has already been appended with a suffix.
 
Is this a problem? If it is maybe we can filter both filters:

 - filter "pre" to just avoid the appending.
 - filter this one to apply the right slug.

Reply:
>  Do you think we can still keep the wp_unique_post_slug filter instead of pre in which case the taxonomy will be there?

IIRC I found that this `get_the_terms` was an issue on the normal hook as well, before I moved over to the pre-* hook.

> If it is maybe we can filter both filters:

If the `pre` filter returns something, it acts as an override and acts as the early return for the `wp_unique_post_slug()` function, so the non-pre filter would never run in that case.

> This will work now but won't scale later when we'll support multi-themes.

🤔   What if we continue to check for the term, but if it doesn't exist fall back to the current theme?

Reply:
> 🤔 What if we continue to check for the term, but if it doesn't exist fall back to the current theme?

This doesn't work either, a REST API can be called to create a template by specifying an explicit theme that is different from the current theme.

--- 
I'm not too concerned about this particular thing at the moment, but I feel like we can make it multi-theme by default, it will avoid us some future struggles.

Reply:
> I'm not too concerned about this particular thing at the moment, but I feel like we can make it multi-theme by default, it will avoid us some future struggles.

Agreed.  While we do have some other sections we will need to adapt to allow multi-theme to work, it would be nice to start thinking about these things now.. and preferably set this up so it would work in that environment. 🤔

Reply:
Just noticing the normal (non-pre) filter has $original_slug as its 6th arg.  So we can probably use this and avoid using the `pre-*` filter.  Although I think that will still have the same behavior and leaves us at the same problem regarding the tax term at this stage.

Reply:
>Just noticing the normal (non-pre) filter has $original_slug as its 6th arg. So we can probably use this and avoid using the pre-* filter.

Tried this out and it can work the same way on the not-pre filter, although it has the same problems with the tax term not being available.  However, we are treating it as an override which is what the `pre-*` filter was meant to do, and doing that with the normal filter just means we are running extra code for no reason since we are overriding it anyways.  With that in mind, sticking with the `pre-*` filter makes the most sense to me.

Regarding how to get this working for a future multi-theme environment, I don't have any 'good' ideas.  The only thing I can think of at this point would be a hacky approach:  force the slug in creation to carry the theme value as well `'name' => 'desired-slug//desired-theme'` and have this unique_post_slug filter function dissect that into using the first half as the slug and the later half as the theme to check against for whether or not we need to append a suffix.  I don't like that solution, and would like to see a better one... but its difficult since this filter doesn't pass any more useful information for the tax query and its too early in the process for `get_the_terms` to carry the corresponding value. 😞

Reply:
Maybe we should leave this problem for later but clarify with a comment?

Reply:
makes sense to me, added.